classes/controller/FrontController.php

```diff
@@ -362,10 +362,14 @@ public function init()
             if ((!$has_currency || $has_country) && !$has_address_type) {
                 if ($has_country && Validate::isLanguageIsoCode($this->context->cookie->iso_code_country)) {
                     $id_country = (int) Country::getByIso(strtoupper($this->context->cookie->iso_code_country));
-                } elseif (Configuration::get('PS_DETECT_COUNTRY') && isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])
-                        && preg_match('#(?<=-)\w\w|\w\w(?!-)#', $_SERVER['HTTP_ACCEPT_LANGUAGE'], $array)
-                        && Validate::isLanguageIsoCode($array[0])) {
-                    $id_country = (int) Country::getByIso($array[0], true);
+                } elseif (Tools::isDetectCountryFromBrowserLanguageAvailable()) {
+                    $languagesAvailable = Tools::languagesAvailable();
+                    Tools::isValidAcceptLanguageAvailable();
+                    if (isset($languagesAvailable[0])) {
+                        $id_country = (int) Country::getByIso($languagesAvailable[0], true);
```

Comment:
If the methods are a bit modified this could be simplified as:
```suggestion
                    if (Tools::isCountryIsoCodeAvailableInHeader()) {
                        $id_country = (int) Country::getByIso(Tools::getCountryIsoCodeFromHeader(), true);
```

Reply:
also applied